libqpdf/QPDF.cc

```diff
@@ -2193,70 +2194,62 @@ QPDF::reserveObjects(QPDFObjectHandle foreign, ObjCopier& obj_copier, bool top)
 
     if (foreign.isIndirect()) {
         QPDFObjGen foreign_og(foreign.getObjGen());
-        if (obj_copier.visiting.find(foreign_og) != obj_copier.visiting.end()) {
-            QTC::TC("qpdf", "QPDF loop reserving objects");
+        if (obj_copier.object_map.count(foreign_og) > 0) {
+            QTC::TC("qpdf", "QPDF already reserved object");
+            if (obj_copier.visiting.count(foreign_og)) {
+                QTC::TC("qpdf", "QPDF loop reserving objects");
+            }
```

Comment:
It appears to me that obj_copier.visiting is unnecessary, since the actual loop avoidance can be achieved by checking the object_map.